code/game/objects/structures/crates_lockers/crates.dm

```diff
@@ -680,3 +680,30 @@
 	log_admin(msg)
 
 #undef RECURSION_PANIC_AMOUNT
+
+/obj/structure/closet/crate/secure/sec_shuttle_ruin
+	name = "locked crate"
+	desc = "The side of the crate has a slot for a keycard to be swiped."
+	can_be_emaged = FALSE
+	var/linked_card_code = "sec_shuttle_ruin"
+
+/obj/structure/closet/crate/secure/sec_shuttle_ruin/allowed(mob/user)
+	if(!user)
+		return
+
+	var/obj/item/card/sec_shuttle_ruin/keycard = user.get_active_hand()
+	if(!keycard)
+		return
+	if(!istype(keycard))
+		return
+
+	if(linked_card_code != keycard.linked_crate_code)
+		return
+	to_chat(user, "<span class='notice'>You swipe [keycard] in [src]'s keycard slot.</span>")
+	return TRUE
+
+/obj/item/card/sec_shuttle_ruin
+	name = "warden's card"
+	desc = "A card used by the warden to unlock their crate."
+	icon_state = "data"
+	var/linked_crate_code = "sec_shuttle_ruin"
```

Comment:
you create a new type and create a new variable both to perform the same exact purpose. 

I recommend just getting rid of this variable and sticking with the typecheck